CMakeLists.txt

```diff
@@ -267,13 +267,16 @@ target_link_libraries(torrent-rasterbar ${Boost_LIBRARIES} ${CMAKE_THREAD_LIBS_I
 #add_definitions(-DBOOST_ASIO_HASH_MAP_BUCKETS=1021  -D__USE_W32_SOCKETS -DWIN32_LEAN_AND_MEAN )
 
 if (WIN32)
-	target_link_libraries(torrent-rasterbar wsock32 ws2_32 Iphlpapi)
+	target_link_libraries(torrent-rasterbar wsock32 ws2_32 Iphlpapi dbghelp)
 	add_definitions(-D_WIN32_WINNT=0x0600)
 	# prevent winsock1 to be included
 	add_definitions(-DWIN32_LEAN_AND_MEAN)
 	if (MSVC)
 		add_definitions(-DBOOST_ALL_NO_LIB)
-		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP") # multicore compilation
+		# for multicore compilation
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP")
```

Comment:
my understanding of what's considered "modern" cmake style is to never define any ABI-affecting flags in libraries, but just to assert that they were set by the top level target. I don't have a strong opinion, and this might make sense. Just curious if you have any opinion on how practical that recommendation would be.

Reply:
I guess this isn't actually a change in this patch anyway

Reply:
What do you exactly mean, to use `target_compile_options()` for this two flags? I can move the `if` statement to root block and use `target_compile_options()` as it is in master, its better than modify this flags by `set()`.

Reply:
this isn't actually a comment on your patch, but just how to use cmake in general. My understanding is that since a target cannot propagate requirements *up* the dependency graph in cmake, there's nothing preventing ABI-incompatible objects from being linked together. for instance, if the target that depends on libtorrent does *not* build with ``/MP`` but still links against libtorrent built with ``/MP``, that would cause an ABI incompatibility and subtle memory corruption.

heh, I figured I should look up ``/MP`` just to be sure, and it turns out I had mixed it up with ``/MD`` (to link against the multi-threaded runtime library). So, my comment is even more abstract now, since it doesn't even apply to this line.

Reply:
Ohh I understand, don't know if /MP somehow affects ABI, I think there will be no problem ( with static or dynamic linkage ) if the lib is compiled with /MP and an app which use this lib isn't compiled with /MP. I can check it later, I looked at [msdn docs](https://msdn.microsoft.com/en-us/library/bb385193.aspx) and it is more aggresive option than I expected, should be checked :)

Reply:
I think the current best practice with cmake is to use the `target_*` commands which _do_ propagate requirements up the dependency graph. Hence the change in master to use those.

Reply:
I was under the impression they just propagate down (i.e. to dependencies of the target, not targets depending on this)

Reply:
As described here https://cmake.org/cmake/help/v3.9/manual/cmake-buildsystem.7.html#target-usage-requirements